components/places/src/bookmark_sync/engine.rs

```diff
@@ -493,6 +494,32 @@ fn apply_remote_items(db: &PlacesDb, scope: &SqlInterruptScope, now: Timestamp)
         }
     ).ok(); // We ignore the error because it's expected, users should not have entries with the same id that have different types
 
+    // We check if any users have any undetected orphaned bookmarks and
+    // log them appropiately
+    let orphaned_count: i64 = db.query_row(
+        "WITH RECURSIVE orphans(id) AS (
+           SELECT b.id
+           FROM moz_bookmarks b
+           WHERE b.parent IS NOT NULL
+             AND NOT EXISTS (
+               SELECT 1 FROM moz_bookmarks p WHERE p.id = b.parent
+             )
+           UNION
+           SELECT c.id
+           FROM moz_bookmarks c
+           JOIN orphans o ON c.parent = o.id
+         )
+         SELECT COUNT(*) FROM orphans;",
+        [],
+        |row| row.get(0),
+    )?;
+
+    if orphaned_count > 0 {
+        log::warn!("Found {} orphaned bookmarks during sync", orphaned_count);
+        breadcrumb!("bookmarks-sync: found local orphaned bookmarks");
+        error_support::report_error!("bookmarks-sync", "orphaned bookmarks {}", orphaned_count);
```

Comment:
@lougeniaC64 I couldn't find an easy way to add telemetry to this output here so I kinda use "sentry" for this. Is there potentially anything wrong here? I know you did some work for this reporting stuff so not sure if i'm missing something.

Reply:
@skhamis That looks right to me but you can also test this by using the [test a-s sentry project](https://mozilla.sentry.io/settings/projects/application-services-test/keys/) to be sure. If you change the sentry DSN in either of the mobile apps it should report to the test project (though I've only done this in iOS). This can be especially useful for double-checking the breadcrumbs.

Reply:
Thanks! I also decided to add a bit of test scaffolding for the `report_error` so we can actually use them in tests for checking things!